Index: Pattern_drawing/pyramid.py
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/Pattern_drawing/pyramid.py b/Pattern_drawing/pyramid.py
new file mode 100644
--- /dev/null	(date 1726157125855)
+++ b/Pattern_drawing/pyramid.py	(date 1726157125855)
@@ -0,0 +1,8 @@
+height = int(input())
+
+for i in range(1, height + 1):
+    for j in range(height - i):
+        print(" ", end="")
+    for j in range(2 * i - 1):
+        print("*", end="")
+    print()
\ No newline at end of file
Index: Pattern_drawing/triangle_numbers.py
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/Pattern_drawing/triangle_numbers.py b/Pattern_drawing/triangle_numbers.py
new file mode 100644
--- /dev/null	(date 1726157428498)
+++ b/Pattern_drawing/triangle_numbers.py	(date 1726157428498)
@@ -0,0 +1,6 @@
+n = int(input())
+
+for row in range(1, n+1):
+    for col in range(1, row + 1):
+        print(col, end="")
+    print("")
\ No newline at end of file
Index: Pattern_drawing/inverted_pyramid.py
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/Pattern_drawing/inverted_pyramid.py b/Pattern_drawing/inverted_pyramid.py
new file mode 100644
--- /dev/null	(date 1726162881869)
+++ b/Pattern_drawing/inverted_pyramid.py	(date 1726162881869)
@@ -0,0 +1,8 @@
+height = int(input())
+
+for row in range(height, 0, -1):
+    for col in range(height - row):
+        print(" ", end="")
+    for col in range(2 * row - 1):
+        print("*", end="")
+    print()
\ No newline at end of file
Index: Pattern_drawing/diamond_numbers.py
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/Pattern_drawing/diamond_numbers.py b/Pattern_drawing/diamond_numbers.py
new file mode 100644
--- /dev/null	(date 1726163398415)
+++ b/Pattern_drawing/diamond_numbers.py	(date 1726163398415)
@@ -0,0 +1,16 @@
+height = int(input())
+
+mid = height // 2 + 1
+
+for i in range(1, mid + 1):
+    for j in range(height - i):
+        print(" ", end="")
+    for j in range(2 * i - 1):
+        print(j + 1, end="")
+    print()
+for i in range(mid - 1, 0, -1):
+    for j in range(height - i):
+        print(" ", end="")
+    for j in range(2 * i - 1):
+        print(j + 1, end="")
+    print()
\ No newline at end of file
Index: Pattern_drawing/mirrored_triangle.py
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/Pattern_drawing/mirrored_triangle.py b/Pattern_drawing/mirrored_triangle.py
new file mode 100644
--- /dev/null	(date 1726163737740)
+++ b/Pattern_drawing/mirrored_triangle.py	(date 1726163737740)
@@ -0,0 +1,8 @@
+n = int(input())
+
+for row in range(1, n+1):
+    for col in range(n - row):
+        print(" ", end="")
+    for col in range(1, row + 1):
+        print("*", end="")
+    print()
Index: Pattern_drawing/star_dash_square.py
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/Pattern_drawing/star_dash_square.py b/Pattern_drawing/star_dash_square.py
new file mode 100644
--- /dev/null	(date 1726164150507)
+++ b/Pattern_drawing/star_dash_square.py	(date 1726164150507)
@@ -0,0 +1,9 @@
+n = int(input())
+
+for row in range(1, n + 1):
+    for col in range(1, n + 1):
+        if (row + col) % 2 == 0:
+            print("*", end='')
+        else:
+            print("-", end='')
+    print()
Index: Pattern_drawing/right-angled_triangle.py
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/Pattern_drawing/right-angled_triangle.py b/Pattern_drawing/right-angled_triangle.py
new file mode 100644
--- /dev/null	(date 1726154832687)
+++ b/Pattern_drawing/right-angled_triangle.py	(date 1726154832687)
@@ -0,0 +1,6 @@
+n = int(input())
+
+for row in range(n+1):
+    for col in range(row):
+        print("*", end="")
+    print("")
\ No newline at end of file
Index: Pattern_drawing/square_with_hollow_center.py
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/Pattern_drawing/square_with_hollow_center.py b/Pattern_drawing/square_with_hollow_center.py
new file mode 100644
--- /dev/null	(date 1726155799542)
+++ b/Pattern_drawing/square_with_hollow_center.py	(date 1726155799542)
@@ -0,0 +1,7 @@
+n = int(input())
+
+for i in range(n):
+    if i == 0 or i == n - 1:
+        print("*" * n)
+    else:
+        print("*" + " " * (n - 2) + "*")
\ No newline at end of file
Index: Pattern_drawing/diamond.py
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/Pattern_drawing/diamond.py b/Pattern_drawing/diamond.py
new file mode 100644
--- /dev/null	(date 1726163317429)
+++ b/Pattern_drawing/diamond.py	(date 1726163317429)
@@ -0,0 +1,8 @@
+height = int(input())
+
+mid = height // 2 + 1
+
+for i in range(1, mid + 1):
+    print(" " * (mid - i) + "*" * (2 * i - 1))
+for i in range(mid - 1, 0, -1):
+    print(" " * (mid - i) + "*" * (2 * i - 1))
\ No newline at end of file
Index: Pattern_drawing/left-angled_triangle.py
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/Pattern_drawing/left-angled_triangle.py b/Pattern_drawing/left-angled_triangle.py
new file mode 100644
--- /dev/null	(date 1726156764866)
+++ b/Pattern_drawing/left-angled_triangle.py	(date 1726156764866)
@@ -0,0 +1,6 @@
+number = int(input())
+
+for i in range(number, 0, -1):
+    for j in range(0, i):
+        print('*', end='')
+    print()
\ No newline at end of file
Index: Pattern_drawing/hollow_square.py
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/Pattern_drawing/hollow_square.py b/Pattern_drawing/hollow_square.py
new file mode 100644
--- /dev/null	(date 1726156903546)
+++ b/Pattern_drawing/hollow_square.py	(date 1726156903546)
@@ -0,0 +1,7 @@
+n = int(input())
+
+for i in range(n):
+    if i == 0 or i == n - 1:
+        print("*" * n)
+    else:
+        print("*" + " " * (n - 2) + "*")
\ No newline at end of file
